queue_block.py

```diff
@@ -109,87 +92,78 @@ def pop(self, grp="null", count=1, reload=False):
             top_n (list): 'Count' signals from the front of the queue.
 
         '''
-
+        count = self.chunk_size
+        reload = self.reload
         # lock the queue we're popping from
-        self._logger.debug("pop: {} {} {}".format(grp,
-                                                  count,
-                                                  reload))
+        self._logger.debug("pop: {} {} {}".format(grp, count, reload))
         lock = self._get_lock(grp)
-        lock.acquire()
-
-        # check out the front of the queue
-        top_n = self.peek(grp, count)
-
-        # only remove the signals if there are signals to remove
-        if top_n:
+        with lock:
+            # check out the front of the queue
+            top_n = self._queues[grp][0:count]
             self._logger.debug(
-                "Removing %d signals from %s_queue" % (count, grp)
-            )
-            self._queues[grp] = self._queues[grp][count:]
+                "Removing %d signals from %s_queue" % (len(top_n), grp))
+            self._queues[grp] = self._queues[grp][len(top_n):]
```

Comment:
Per this SO post (http://stackoverflow.com/a/1352907/1222076) we should do this:

```python
self._queues[grp][:] = self._queues[grp][len(top_n):]
```